### ISSUE TITLE
iOS still stop at Installing gRPC-Core (1.3.0) step

### ISSUE THREAD

**Issue by zfha** [GoogleCloudPlatform/ios-docs-samples#43]
```
ProductName:	Mac OS X
ProductVersion:	10.12.4
BuildVersion:	16E195
pod-version: 1.2.1
path: /Users/xxx/Desktop/grpc/examples/objective-c/helloworld
```
when I run `pod install`. It stop at  `Installing gRPC-Core (1.3.0)` step;
```
➜  helloworld git:(master) pod install
Analyzing dependencies
Fetching podspec for `HelloWorld` from `.`
Downloading dependencies
Installing !ProtoCompiler (3.2.0)
Installing !ProtoCompiler-gRPCPlugin (1.3.0)
Installing BoringSSL (8.2)
Installing HelloWorld (0.0.1)
Installing Protobuf (3.3.0)
Installing gRPC (1.3.0)
Installing gRPC-Core (1.3.0)
```
Help me !

**Comment by canny09** [comment]
I also has this issue... Did you fixed it?

**Comment by zfha** [comment]
In China？
1. I used Shadowsocks（proxy software）
2. Set .gitconfig proxy
3. This step need long time. About 10+ minutes.

`.gitconifg` 
```
[http]
	proxy = socks5://127.0.0.1:1080
[https]
	proxy = socks5://127.0.0.1:1080
```

**Comment by canny09** [comment]
yes, in China.  I also used the Shadowsocks, but nothing changed. Thanks, I will try to config the git proxy, hope it works.

**Comment by GrEg00z** [comment]
Same problem here !

**Comment by AwaisFayyaz** [comment]
Experiencing the same issue. any solution to this ?

**Comment by AwaisFayyaz** [comment]
This actually takes some time (around 30 minutes) so let this thing install

**Comment by notmedia** [comment]
Same thing with gRPC-Core (1.28.2)

**Comment by ghasemikasra39** [comment]
same here with Installing gRPC-Core (1.28.2). Any solution ?

**Comment by AlexAlexandre** [comment]
> same here with Installing gRPC-Core (1.28.2). Any solution ?

man, I had the same problem, but reading the comments I saw that just wait a little, here it took about 10 minutes.

**Comment by wikramhwt** [comment]
I am also facing the same issue and completely lost my whole day on solving this issue
i just add these two files in my pod file before these two files everything was working fine
```
pod 'Firebase/Core'
pod 'Firebase/Firestore'
```
every thing goes well but when it comes to this one 
`Installing gRPC-Core (1.28.2)`
it goes stuck
kindly help me

**Comment by iqbalmineraltown** [comment]
I'm also stuck on `Installing gRPC-Core (1.28.2)`
Already running for 55 minutes before I canceled it

**Comment by OmarUnity** [comment]
I resolved it by deleting all the cache related to gRPC at `/Users/<user>/Library/Caches/CocoaPods/Pods/Release/` and .`../Specs/Release/`, and running `pod install` again.

**Comment by ImHamzaAnwer** [comment]
did anyone find the solution ?

**Comment by nitishk72** [comment]
## Solution

It is dowing grpc core and it's submodule and it will take time. The file size is greater than or around 1 GB. So sit back and relax. If you would like to learn more then you can read my article about the same problem and the solution.

This article only discusses the problem and the solution "Why it stuck and how to get rid of this?"

https://www.nstack.in/blog/grpc-ios-stuck/

**Comment by MuhammadFaizan980** [comment]
wait and see is the only logical solution as it is around 1GB

**Comment by sultanmyrza** [comment]
checkout @nitishk72 article at https://www.nstack.in/blog/grpc-ios-stuck/ worth reading!

My case: flutter, firestore.
I did `pod install` and stuck at  gRPC-Core
so I cancel it 
`flutter clean`
and run agian
`pod install --verbose`
you still have to wait but at leas you will see whats happening!

# My internet connection
## DOWNLOAD Mbps 43.90
## UPLOAD Mbps 20.87
# took me aroud 30 mins

**Comment by OrtalPozniak** [comment]
I had the same issue, if you run pod install --verbose
it still takes a lot of time but you will be able to see the progress.

**Comment by mgfjx** [comment]
I got this `Failed to clone 'third_party/googleapis' a second time, aborting` even if use shadowsocks.

**Comment by Luohuhu** [comment]
> I got this `Failed to clone 'third_party/googleapis' a second time, aborting` even if use shadowsocks.

I got the same problem, can you tell me how to fix it?

**Comment by Shaildodia** [comment]
Just wait for little while. For me it took around 8 to 10 minutes to install it.

**Comment by Luohuhu** [comment]
> Just wait for little while. For me it took around 8 to 10 minutes to install it.

for me it took more than an hour so I kill that process

**Comment by gurayaharsh** [comment]
Same issue. Been waiting for over 30 mins at this point but I guess theres no solution other than waiting.

**Comment by mesheilah** [comment]
I have faced the same issue, although I have a very fast fiber optic connection (130 mbps download speed), but it took about 5 hours for the `gRPC-Core` to fully cloned with `pod install`

**Comment by carloliaci** [comment]
**Bad Network Adapter!**
Unbelievable, after many days I fixed the gRPC-Core issue! My problem was because of the wired network adapter. I used an Atolla Usb/ethernet apple adapter (Amazon). I switched to wifi connection and I correctly installed all the pods! I will burn the adapter and I will buy an original apple ethernet. I hope that it will help you. :)

**Comment by oti-adjei** [comment]
It does a long time. Im in Ghana West Africa and it took me almost an hour and half on my m1. Note also though that your internet should be stable and of decent speed

**Comment by nihatdinc** [comment]
When I read this thread my gRPC-Core has been downloaded, lol. Take a break and just wait.

**Comment by zzyogeshrathore** [comment]
Installing gRPC-C++ (1.28.2)

Installing gRPC-Core (1.28.2)



Actually its a large file so it will take some time in my case it took 45 min with mobile data. so keep patience if you have fast internet definitely it will take less time.

**Comment by yogeshrathore123** [comment]
Installing gRPC-C++ (1.28.2)

Installing gRPC-Core (1.28.2)



Actually its a large file so it will take some time in my case it took 45 min with mobile data. so keep patience if you have fast internet definitely it will take less time.

**Comment by h-bomb** [comment]
`Installing gRPC-Core (1.28.2)` took over an hour for me, but it did finally finish. I'm in California with about a 50 Mbps download rate.

**Comment by allooshcode** [comment]
it is absolutely a network issue. please try to get more speed internet.

**Comment by CliffLeopard** [comment]
It's been five years and there's still no solution?

**Comment by mubashardev** [comment]
> It's been five years and there's still no solution?

The solution is nothing but just waiting for it to get finished. It's about 1GB and let it take time to download and install :)

**Comment by ndukayames** [comment]
omo

**Comment by Fawaz441** [comment]
> omo

aswear, this thing takes too long.😭

**Comment by GwamakaCharles** [comment]
whoever created this,  thought everyone who will use it works at Google and they'll have 10Gbps of internet speed...

**Comment by slagathor2094** [comment]
> It's been five years and there's still no solution?

It's been five years, and it still hasn't finished downloading on my machine.

**Comment by webdiego** [comment]
Same

**Comment by GwamakaCharles** [comment]
Turns out this is an internet problem....! You don't really need an extremely fast internet, just 100kb/s will do. You also need some patience...😀

**Comment by ildaneta** [comment]
The problem is that it's too long and you have to be patience. 😊

**Comment by 107ab0703** [comment]
> 我通過刪除所有與 gRPC 相關的緩存來解決它`/Users/<user>/Library/Caches/CocoaPods/Pods/Release/`。`../Specs/Release/`，然後`pod install`再次運行。

After deleted caches, I just run 10min !!!!! Thanks 🥳 (My mac is M1)

**Comment by gizemay96** [comment]
> I resolved it by deleting all the cache related to gRPC at `/Users/<user>/Library/Caches/CocoaPods/Pods/Release/` and .`../Specs/Release/`, and running `pod install` again.

Thank you so much

**Comment by khuznain99** [comment]
You may like this: https://github.com/invertase/firestore-ios-sdk-frameworks - a separate package that will cut the need for the gRPC dependency out as a direct dependency and install it pre-compiled for you which saves an incredible amount of time.

**Comment by oti-adjei** [comment]
Smart solution. With this though does it crosscheck to see versions are compatible  with other packages etc. before compiling?

**Comment by kelle62819** [comment]
Why is this code so bloated? 1GB is ridiculous

**Comment by EchoEllet** [comment]
In case of flutter open the terminal in the directory of your flutter project, `flutter clean` then `flutter pub get`
cd to ~/Library/Caches
and delete CocoaPods
by simply rm -rf CocoaPods/

then cd back to the directory of your flutter project and then cd to the ios project inside it
pod install --verbose
It's a large file you will have to wait but at least you can see the progress and it shouldn't be stuck forever

**Comment by xavit** [comment]
Just do a Complete cleanup (pod reset)
`rm -rf ~/Library/Caches/CocoaPods
rm -rf Pods
rm -rf ~/Library/Developer/Xcode/DerivedData/*
pod deintegrate
pod setup
pod install`
